src/compiler/sys/resolve/resolve-module-sync.ts

```diff
@@ -139,9 +139,12 @@ export const createCustomResolverSync = (
       const fsFilePath = normalizeFsPath(p);
       try {
         return sys.realpathSync(fsFilePath);
-      } catch (realpathErr) {
-        if (realpathErr.code !== 'ENOENT') {
-          throw realpathErr;
+      } catch (realpathErr: unknown) {
+        if (realpathErr instanceof Object && 'code' in realpathErr) {
+          // @ts-ignore: we've determined 'code' is in the prototype chain, but TS isn't happy about the access
```

Comment:
FYI: This is because `in` isn't a type guard: microsoft/TypeScript#21732. One option would be to implement a type guard function for [`NodeJS.ErrnoException`]():

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/globals.d.ts#L183-L188

Reply:
Good call, impl in f0fb959c7..b88604bad